Warning: inline variables are a C++17 extension

## Bug description
- Warning: inline variables are a C++17 extension

## Version info
- Version or branch: dev (2021-06-29)
- Qt version: 5.14.2
- Platform: macOS 10.15.6 Mojave and Ubuntu 20.04 64-bit

## Message
```
In file included from ../../autonomx/autonomx/Generator.h:27:
../../autonomx/autonomx/GeneratorMeta.h:51:12: warning: inline variables are a C++17 extension [-Wc++17-extensions]
```

## Technical details
- We can probably update the version of C++ required to 17, in the .pro file.

--- a/autonomx/autonomx.pro
+++ b/autonomx/autonomx.pro
@@ -3,7 +3,7 @@ QT += core
 QT += qml
 QT += gui
 # QT += serialport
-CONFIG += c++11
+CONFIG += c++17
 
 # registers custom QML types
 CONFIG += qmltypes

